src/Background/Portal.vala

```diff
@@ -135,13 +138,12 @@ public class Background.Portal : Object {
          */
         var app_info = new DesktopAppInfo (_app_id + ".desktop");
         if (app_info == null) {
-            _app_id = string.joinv ("-", commandline).replace ("--", "-");
+            _app_id = string.joinv ("-", commandline).replace ("--", "-").replace ("--", "-");
         }
 
-        var path = Path.build_filename (Environment.get_user_config_dir (), "autostart", _app_id + ".desktop");
-
+        var file = File.new_build_filename (Environment.get_user_config_dir (), "autostart", _app_id + ".desktop");
```

Comment:
Only out of curiosity: What's the advantage here?

Reply:
GLib recommends that we should use the asynchronous versions of the methods in asynchronous code. In this case, it's reduce the chances that we slow or block others methods and others portals.